File: bin/people-array.js
'use strict';

let loadPeople = require('../lib/people.js');

loadPeople().then((people) => {
  let count = people.reduce((memo, person) => {
    return memo + 1;
  }, 0);

  const isFemale = (person) => {
    return person.gender === "f";
  }

  let female = people.filter(isFemale(person)).reduce((person) => {
    return memo + 1;
  }, 0);

  console.log(`female: ${female}, total: ${total}`);
  // console.log(`count: ${count}, length: ${people.length}`);

  // Jeff Horn is 30 years old.
  // people.forEach((person) => {
  //   console.log(`${person.given_name} ${person.surname} is ${person.age()} years old.`);
  // });
}).catch((error) => console.log(error));

File: bin/people_array.rb
require_relative '../lib/people.rb'

people = People.new.people
total = people.count.to_f

female = people.select { |person| person.gender == 'f' }.length
female_percentage = female / total
p "female: #{female}, total: #{total}, female_percentage: #{female_percentage}"

older_people = people.select { |person| person.age > 25 }.length
p "older people: #{older_people}"

younger_people = people.select { |person| person.age < 25 }.length
p "younger people: #{younger_people}"

letter_matches = people.select { |person| person.given_name[0] == person.surname[0] }.length
p "letter_matches: #{letter_matches}"

letter_matches_people = people.select { |person| person.given_name[0] == person.surname[0] }
# I had this here as a check
# puts "letter_matches_people: #{letter_matches_people}"

average_age = people.reduce(0) { |result, element| result + element.age } / total
p "avg age: #{average_age}"
